### PR TITLE
[flutter_local_notifications] Fix pub versioning error in Flutter 1.22

### PR DESCRIPTION
**Pull request by acoutts**
This fixes the issue here: https://github.com/MaikuB/flutter_local_notifications/issues/782

### CODE REVIEW

File: flutter_local_notifications/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# [1.4.4+5]
+
+* [Android] Fix issue [782](https://github.com/MaikuB/flutter_local_notifications/issues/782) by removing unused dependency to the platform lib which fixes the pub versioning error in Flutter 1.22.
```
Comment (MaikuB) [inline flutter_local_notifications/CHANGELOG.md:3]: Changelog entry will need tweaking as the dependency is actually used (it's imported in `flutter_local_notifications_plugin.dart`) but looks like it's now pulled as a transitive dependency. I'm assuming that it's the Flutter SDK that is pulling it in

**Review by MaikuB — CHANGES_REQUESTED** [review]
Thanks for the PR. Looks good to me but the changelog entry will need a little tweaking

File: flutter_local_notifications/CHANGELOG.md (hunk 1 above)
Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: In that case do we need to mark this a new major version as trying to install it prior to Flutter 1.22 might not work then?

Comment (MaikuB) [reply flutter_local_notifications/CHANGELOG.md:3]: Good question, I've only tried it with Flutter 1.2.0. I'll need to be tested on 1.12, if it still works then should be good to merge the PR in after amending the changelog. However, it a version bump is needed then I would suggest that the PR goes to the other branch of the repo that is already being used for the 1.5.0 beta that I intend to release soon. I'm waiting to test an upcoming daylight savings changes that takes on this weekend before I make it a stable release

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: Ah! I think it makes sense to just make this part of 1.5.0 then. It's just a simple one line fix to remove the dependency.

What's the best way to make this part of that release?

Comment (MaikuB) [reply flutter_local_notifications/CHANGELOG.md:3]: I believe the best way is close this PR and make it that you do your PR based off the `time_zone_macos_support` branch. I see you've been doing PRs from your own master branch. One tip I learnt recently to create your own branches on your fork when submitting a PR. That can make it easier to keep your fork of master up to date so it exactly matches the original repo's master branch

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: Thanks for that! Alright i'll do that.

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: So to confirm you want me to fork off of `time_zone_macos_support` to my own github, create a branch called `fix-782` to reference my issue number, and create a PR to merge my branch into the `time_zone_macos_support` one here?

Comment (MaikuB) [reply flutter_local_notifications/CHANGELOG.md:3]: Only if it's necessary. If the PR works with Flutter 1.12 then it's not needed. If it doesn't then the PR would also need to bump the minimum supported version of Flutter.

You already have a fork of this repo and that should already have the `time_zone_macos_support` for you to branch off from provided you've kept your fork up to date.

Comment (MaikuB) [reply flutter_local_notifications/CHANGELOG.md:3]: Should I add that I would suggest testing with Flutter 1.12 first if you can. The Flutter website does have an archive of older SDKs up on there

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: I just checked out stable branch and notice it's on 1.20.2 - curious why target 1.12 still if the stable is 1.20 now?

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: Ah right - minimum supported version, I see.

Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: I tried it out on 1.12.13+hotfix.9 (same as in the min version), and it seems that platform is pulled in transitively here too, so I think it's safe to do this on the current version / sdk support.

**Comment by acoutts** [comment]
Closing as I moved branches.

File: flutter_local_notifications/CHANGELOG.md (hunk 1 above)
Comment (acoutts) [reply flutter_local_notifications/CHANGELOG.md:3]: I'm actually not able to run the example app from master branch on 1.12.13:
```
2020-09-02 08:25:24.000368-0400 Runner[78869:5970246] Failed to find assets path for "Frameworks/App.framework/flutter_assets"
2020-09-02 08:25:24.066855-0400 Runner[78869:5970804] [VERBOSE-2:engine.cc(127)] Engine run configuration was invalid.
2020-09-02 08:25:24.067021-0400 Runner[78869:5970804] [VERBOSE-2:shell.cc(437)] Could not launch engine with configuration.
2020-09-02 08:25:24.096888-0400 Runner[78869:5970821] flutter: Observatory listening on http://127.0.0.1:61872/Rr3T8tYt9GI=/
```

It does work fine on the latest stable 1.20 though. It appears the minimum supported SDK needs bumped regardless of the change here.

**Comment by MaikuB** [comment]
Did you check Android? I believe the issue you're seeing is more to do with the Xcode project having been migrated as per https://flutter.dev/docs/development/ios-project-migration

**Comment by acoutts** [comment]
Ah android looks fine, I understand. I just opened https://github.com/MaikuB/flutter_local_notifications/pull/785 to try this again.

**Pull request closed without merging**